catroid/src/main/java/org/catrobat/catroid/devices/mindstorms/nxt/Command.java

```diff
@@ -63,4 +66,19 @@ public int getLength() {
 	public byte[] getRawCommand() {
 		return commandData.toByteArray();
 	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (!(obj instanceof Command)) {
+			return false;
+		}
+
+		Command command = (Command) obj;
+		return Arrays.equals(commandData.toByteArray(), command.commandData.toByteArray());
+	}
+
+	@Override
+	public int hashCode() {
+		return super.hashCode() * TAG.hashCode();
```

Comment:
is this on purpose? "super" here is going to be plain java "object", since this only implements a interface and does not extend anything. (so im unsure what super.hashCode() really produces here, and i doubt its going to fulfill the requirements for hashCode (see java doc))

when i generate a hash method in android studio, i get:

	@Override
	public int hashCode() {
		return Objects.hash(commandData);
	}

which makes sense.

hashCode has to:
- return the same integer on two objects that are equal according to equals(Object obj2)
- not necessarily return different integers if two objects are not equal according to equals(Object obj), but its a good idea because this behaviour could lead to really unexpected behaviour when the object is used in any kind of hashmap (also any kind of Set and some other collection objects because those use hashCode() )

Reply:
You are right, this line makes more sense.
I've copied the other line from Sprite.
Should I refactor it this there to?